embedded/src/main/java/org/wildfly/core/embedded/LoggerContext.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.core.embedded;
+
+import static org.jboss.logging.Logger.Level.WARN;
+
+import org.jboss.logging.Logger;
+import org.jboss.modules.Main;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleClassLoader;
+import org.jboss.modules.ModuleLoadException;
+import org.jboss.modules.ModuleLoader;
+import org.jboss.modules.log.ModuleLogger;
+import org.jboss.modules.log.NoopModuleLogger;
+import org.wildfly.core.embedded.logging.EmbeddedLogger;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+/**
+ * A context for the logging environment.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+class LoggerContext implements Context {
+    private static final String MODULE_ID_LOGGING = "org.jboss.logging";
+
+    private final ModuleLoader moduleLoader;
+
+    private volatile ModuleLogger current;
```

Comment:
just a note to myself: `current` is the ModuleLogger *before activation* (using the name `current` is a bit misleading when you are in the `restore` method).

Reply:
I'll change the name as that does make sense. I'll just name it `toRestore` or something similar.